.github/workflows/build_and_test.yml

```diff
@@ -10,38 +10,89 @@ jobs:
     strategy:
       matrix:
         os: [macos-10.15, ubuntu-latest]
+        netcdf_version: [4.7.4]
         compiler: [gcc-9]      
 
     steps:
+
     - uses: actions/checkout@v2
+
     - name: install-dependencies
       run: |
         if [[ ${{ matrix.os }} == "ubuntu-latest" ]]; then
           sudo apt-get update
           sudo apt-get install libmpich-dev
-          sudo apt-get install libnetcdf-dev libnetcdff-dev netcdf-bin pkg-config
+          sudo apt-get install netcdf-bin pkg-config
```

Comment:
what is `netcdf-bin` package providing here if you are building netcdf in the action?

Reply:
```
$ apt-cache show netcdf-bin
Package: netcdf-bin
Architecture: amd64
Version: 1:4.7.3-1
Priority: optional
Section: universe/science
Source: netcdf
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian GIS Project <pkg-grass-devel@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 550
Depends: libc6 (>= 2.14), libhdf5-103, libnetcdf15 (>= 4.6.2)
Filename: pool/universe/n/netcdf/netcdf-bin_4.7.3-1_amd64.deb
Size: 191060
MD5sum: 1e51605d1fdf28d76872495ad7f42571
SHA1: 7660c3087ac19814b406e2deaad258ddd711bf43
SHA256: e55a34a0d7f550de2f00655cff0061534e8606104de6ac492585d2e601f35593
Homepage: http://www.unidata.ucar.edu/software/netcdf/
Description-en: Programs for reading and writing NetCDF files
 Contains the programs ncdump and ncgen which convert NetCDF
 files to ASCII and back, respectively. NetCDF (network Common Data
 Form) is an interface for scientific data access and a
 freely-distributed software library that provides an implementation
 of the interface.  The netCDF library also defines a
 machine-independent format for representing scientific data.
 Together, the interface, library, and format support the creation,
 access, and sharing of scientific data.
Description-md5: 1f237fd4e802524fad5b1199a6270d85
```

Reply:
right. What is being used from netcdf-bin in this repo?
installing netcdf-bin will bring down libnetcdf, libhdf5.

Reply:
@aerorahul does the netcdf build provide `ncdump` and `ncgen`?

Reply:
and, does it even need those?

Reply:
@aerorahul want me to just get rid of `netcdf-bin` ?

Reply:
If you are building and installing netCDF then you should not also install it from binary. One or the other...

Reply:
yes it does provide it, but I don't see it being used anywhere here.
Please remove `netcdf-bin` from the action as well as `pkg-config`.

Reply:
Also libnetcdf, since it is already being built and installed.

Reply:
I removed it and pkg_config.